packages/flutter_blue_plus_platform_interface/lib/src/bluetooth_msgs.dart

```diff
@@ -352,6 +361,7 @@ class BmBluetoothDescriptor {
       remoteId: DeviceIdentifier(json['remote_id']),
       serviceUuid: Guid(json['service_uuid']),
       characteristicUuid: Guid(json['characteristic_uuid']),
+      characteristicIndex: int.parse(json['characteristic_index']),
```

Comment:
Could we use an `int` instead of a `String` when serializing and deserializing the index?

```suggestion
      characteristicIndex: json['characteristic_index'],
```

https://docs.flutter.dev/platform-integration/platform-channels#codec